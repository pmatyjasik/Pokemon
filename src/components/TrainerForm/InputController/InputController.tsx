import { Box } from "@mui/material";
import { Input, InputProps } from "@/components/Input/Input";
import { Label } from "@/components/Label/Label";
import { ErrorHint } from "@/components/ErrorHint/ErrorHint";

export interface InputControllerProps {
    label: string;
    placeholder?: string;
    error?: string;
    inputProps?: Omit<InputProps, "placeholder" | "type">;
}

export const InputController = ({ label, placeholder, error, inputProps }: InputControllerProps) => {
    return (
        <Box display='flex' flexDirection='column' gap={0.5}>
            <Label label={label} />
            <Input placeholder={placeholder || label} type='input' {...inputProps} />
            <ErrorHint error={error} />
        </Box>
    );
};
