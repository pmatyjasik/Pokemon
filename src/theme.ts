"use client";

import { createTheme, Shadows } from "@mui/material/styles";
import { ThemeOptions } from "@mui/material/styles";
import { NextFont } from "next/dist/compiled/@next/font";
import localFont from "next/font/local";

const IbmVga: NextFont = localFont({
    src: "./fonts/IBM_VGA.woff",
    display: "swap",
    style: "normal",
    weight: "400",
});

const themeOptions: ThemeOptions = {
    palette: {
        mode: "light",
        background: {
            default: "rgba(255, 255, 255, 1)",
        },
        primary: {
            main: "rgba(151, 71, 255, 1)",
            dark: "rgba(113, 53, 191, 1)",
            light: "rgba(151, 71, 255, 0.25)",
        },
        grey: {
            100: "rgba(42, 42, 42, 1)",
            200: "rgba(127, 127, 127, 1)",
            300: "rgba(228, 228, 228, 1)",
            400: "rgba(238, 238, 238, 1)",
        },
        error: {
            main: "rgba(255, 78, 78, 1)",
        },
        action: {
            disabled: "rgba(0, 0, 0, 0.2)",
        },
    },
    spacing: 4,
    typography: {
        fontFamily: IbmVga.style.fontFamily,
        fontWeightMedium: 400,
        fontSize: 16,
        h6: {
            fontSize: "12px",
        },
        caption: {
            fontSize: "10px",
        },
    },
    shape: {
        borderRadius: 2,
    },
    shadows: [
        "none",
        "0px 0px 0px 4px rgba(151, 71, 255, 0.25)",
        "0px 4px 10px 2px rgba(0, 0, 0, 0.1)",
        ...Array(20).fill("none"),
    ] as Shadows,
};

export const theme = createTheme(themeOptions);
