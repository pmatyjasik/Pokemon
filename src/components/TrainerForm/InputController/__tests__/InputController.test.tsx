import { cleanup, render, screen } from "@testing-library/react";
import { ThemeProvider } from "@mui/material/styles";
import { theme } from "@/theme";
import { InputController, InputControllerProps } from "@/components/TrainerForm/InputController/InputController";

const renderComponent = (props: InputControllerProps) => {
    return render(
        <ThemeProvider theme={theme}>
            <InputController {...props} />
        </ThemeProvider>
    );
};

const label = "Test label";
const placeholder = "Test placeholder";

const defaultProps: InputControllerProps = {
    label,
    placeholder,
};

describe("InputForm Component", () => {
    afterEach(() => {
        cleanup();
    });

    it("Matches DOM Snapshot", () => {
        const { asFragment } = renderComponent(defaultProps);

        expect(asFragment()).toMatchSnapshot();
    });

    it("renders the Label component with the correct label", () => {
        renderComponent(defaultProps);

        const labelElement = screen.getByTestId("label");

        expect(labelElement).toHaveTextContent(label);
    });

    it("renders the Input component with the correct placeholder", () => {
        renderComponent({ ...defaultProps });

        const inputElement = screen.getByPlaceholderText(placeholder);

        expect(inputElement).toBeInTheDocument();
    });

    it("renders the Input component with the label as placeholder when placeholder is not provided", () => {
        renderComponent({ ...defaultProps, placeholder: undefined });

        const inputElement = screen.getByPlaceholderText(label);

        expect(inputElement).toBeInTheDocument();
    });
});
