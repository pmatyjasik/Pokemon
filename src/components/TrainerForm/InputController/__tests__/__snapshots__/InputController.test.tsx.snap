// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputForm Component Matches DOM Snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-bc27n2"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-animated MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-animated css-8uvcxz-MuiFormLabel-root-MuiInputLabel-root"
      data-testid="label"
    >
      Test label
    </label>
    <div
      class="MuiFormControl-root MuiTextField-root css-e5ab0l-MuiFormControl-root-MuiTextField-root"
    >
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl css-pk2gai-MuiInputBase-root-MuiOutlinedInput-root"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          id=":r0:"
          placeholder="Test placeholder"
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              â€‹
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
